### PR TITLE
Wip mail embed

### PR DESCRIPTION
**Pull request by kteb**
First trial implementation for fixing the following issue:  #1040

### CODE REVIEW

File: mail/message.go [hunk 1]
```diff
@@ -32,6 +33,7 @@ type Attachment struct {
 	Name        string
 	Reader      io.Reader
 	ContentType string
+	Embeded     bool
```
Comment (robbyoconnor) [inline mail/message.go]: `Embedded`

File: mail/message.go [hunk 2]
```diff
@@ -72,6 +74,19 @@ func (m *Message) AddAttachment(name, contentType string, r io.Reader) error {
 		Name:        name,
 		ContentType: contentType,
 		Reader:      r,
+		Embeded:     false,
```
Comment (robbyoconnor) [inline mail/message.go]: `Embedded`

File: mail/message.go [hunk 3]
```diff
@@ -72,6 +74,19 @@ func (m *Message) AddAttachment(name, contentType string, r io.Reader) error {
...
+	return nil
+}
+
+//AddEmbeded adds the attachment to the list of attachments
+// the Message has and uses inline instead of attachement property.
+func (m *Message) AddEmbeded(name string, r io.Reader) error {
+	m.Attachments = append(m.Attachments, Attachment{
+		Name:    name,
+		Reader:  r,
+		Embeded: true,
```
Comment (robbyoconnor) [inline mail/message.go]: `Embedded`

File: mail/smtp_sender_test.go [hunk 4]
```diff
@@ -36,6 +36,8 @@ func TestSendPlain(t *testing.T) {
 	m.Bcc = []string{"secret@other.com"}
 
 	m.AddAttachment("someFile.txt", "text/plain", bytes.NewBuffer([]byte("hello")))
+	m.AddAttachment("otherFile.txt", "text/plain", bytes.NewBuffer([]byte("bye")))
+	m.AddEmbeded("test.jpg", bytes.NewBuffer([]byte("not a real image")))
```
Comment (robbyoconnor) [inline mail/smtp_sender_test.go]: `addEmbedded()`

**Comment by kteb** [comment]
Is something still missing ?

**Comment by markbates** [comment]
Looks great. Thanks!

**Pull request merged**